lib/go-atscfg/sslservernamedotyaml.go

```diff
@@ -342,22 +342,22 @@ func dsUsesServer(
 	ds *DeliveryService,
 	server *Server,
 	dss []DeliveryServiceServer,
-	nameTopologies map[TopologyName]tc.Topology,
-	cacheGroups map[tc.CacheGroupName]tc.CacheGroupNullable,
+	nameTopologies map[TopologyName]tc.TopologyV5,
+	cacheGroups map[tc.CacheGroupName]tc.CacheGroupNullableV5,
 	serverCapabilities map[int]map[ServerCapability]struct{},
 	dsRequiredCapabilities map[int]map[ServerCapability]struct{},
 ) (bool, error) {
-	if ds.XMLID == nil || *ds.XMLID == "" {
+	if &ds.XMLID == nil || ds.XMLID == "" {
```

Comment:
The `&ds.XMLID == nil` check can be removed.

Reply:
removed